src/printer.js

```diff
@@ -2764,6 +2768,11 @@ function shouldPrintSameLine(node) {
     type === "UnaryExpression";
 }
 
+function isFlowNodeStartingWithDeclare(node, options) {
+  const nodeSource = options.originalText.slice(0, util.locStart(node));
```

Comment:
Okay, last thing I promise :p

Would be nice to return early if the parser is not flow. This way we don't need to compute the slice.

```js
if (option.parser !== "flow") {
  return false;
}

return options.originalText
  .slice(0, util.locStart(node))
  .match(/declare\s*$/);
```